### PR TITLE
[AzureMonitorExporter] Add `ApplicationInsightsSampler` to the exporter

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
* Added `ApplicationInsightsSampler` to the exporter.
* Added `SamplingRatio` property to customize the sampling rate in both Azure Monitor Exporter and Azure Monitor Distro.
* Removed reference to `OpenTelemetry.Extensions.AzureMonitor`.
* Updated tests and demo.

**TODO**

* Remove `ApplicationInsightsSampler` from OpenTelemetry .NET Contrib.
* Deprecate `OpenTelemetry.Extensions.AzureMonitor` NuGet package.
* Update public document.

### CODE REVIEW

**Review by vishweshbankwar — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/AzureMonitorOptions.cs [hunk 1]
```diff
@@ -36,6 +36,13 @@ public class AzureMonitorOptions : ClientOptions
         /// </summary>
         public bool DisableOfflineStorage { get; set; }
 
+        /// <summary>
+        /// Gets or sets the ratio of telemetry items to be sampled. The value must be between 0.0F and 1.0F, inclusive.
+        /// For example, specifying 0.4 means that 40% of traces are sampled and 60% are dropped.
+        /// The default value is 1.0F, indicating that by default, all telemetry items are sampled.
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/AzureMonitorOptions.cs:42]: Don't need the word "default" twice in the same sentence.

Could write this as "The default value is 1.0F, indicating that 100% of telemetry items are sampled."

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs [hunk 2]
```diff
@@ -30,6 +30,13 @@ public class AzureMonitorExporterOptions : ClientOptions
         /// </summary>
         public TokenCredential? Credential { get; set; }
 
+        /// <summary>
+        /// Gets or sets the ratio of telemetry items to be sampled. The value must be between 0.0F and 1.0F, inclusive.
+        /// For example, specifying 0.4 means that 40% of traces are sampled and 60% are dropped.
+        /// The default value is 1.0F, indicating that by default, all telemetry items are sampled.
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterOptions.cs:36]: same

**Review by TimothyMothra — APPROVED** [review]

File: eng/Packages.Data.props [hunk 3]
```diff
@@ -122,7 +122,6 @@
     <!-- OpenTelemetry dependency approved for Azure.Monitor.OpenTelemetry.Exporter package only -->
     <PackageReference Update="OpenTelemetry" Version="1.4.0" />
     <PackageReference Update="OpenTelemetry.Exporter.InMemory" Version="1.4.0" />
-    <PackageReference Update="OpenTelemetry.Extensions.AzureMonitor" Version="1.0.0-beta.4" />
```
Comment (cijothomas) [inline eng/Packages.Data.props:125]: what is the reasoning behind this change?

Comment (TimothyMothra) [reply eng/Packages.Data.props:125]: Plan is to deprecate the package and move those changes here.
I'll ping @rajkumar-rangaraj to discuss with you.

Comment (rajkumar-rangaraj) [reply eng/Packages.Data.props:125]: The `OpenTelemetry.Extensions.AzureMonitor` package has been customized specifically for the Azure Monitor Exporter and is exclusively used by this project. In order to reduce maintenance overhead and simplify configuration for customers, we have moved the implementation to this repository. As part of this change, we plan to remove the code from the contrib repository and deprecate the `OpenTelemetry.Extensions.AzureMonitor` NuGet package.

**Pull request merged**